src/PPN-microbench/cpu_frequency.cpp

```diff
@@ -1,50 +1,77 @@
 #include <PPN-microbench/cpu_frequency.hpp>
 
-CPUFrequency::CPUFrequency(std::string name, int nbIterations) : Microbench(name, nbIterations)  {
-    nbThreads = std::thread::hardware_concurrency() - 1;
-    measures.reserve(nbThreads * nbIterations);
-    benchTimes.reserve(nbThreads * nbIterations);
+CPUFrequency::CPUFrequency(int nbMeasures) : Microbench("CPU Frequency", 9999999){
+    this->nbMeasures = nbMeasures;
+    Context context = Context::getInstance();
+    nbCores = context.getCpus();
+    measures = std::make_unique<double[]>(nbMeasures * ((nbCores * (nbCores + 1)) / 2));
 }
 
 CPUFrequency::~CPUFrequency() {}
 
-void CPUFrequency::executeBench(int id) {
-    u64 startMeasure = rdtsc();
-    std::chrono::steady_clock::time_point startTime = std::chrono::steady_clock::now();
-    int cpt;
-    for (int i = 0; i < INT32_MAX; i++) {
-        cpt++;
+void CPUFrequency::executeAdds() {
```

Comment:
Inline this function if it does not perform the time measurement, jumps are costly!

Reply:
It's done :D